Frontend/js/raytracing.js

```diff
@@ -1,125 +1,234 @@
+		x = x + (size / 2 );
+		y = y - (size / 2);
+		ctx.lineTo(x + size, y );			
+		x = x + size;
+		ctx.lineTo(x - (size / 2 ), y + (size / 2) );			
+		ctx.closePath();
+		ctx.fillStyle = color;
+		ctx.stroke();
+		ctx.fill();
 			
-			ctx.beginPath();
-			ctx.moveTo(x, y);
-			ctx.lineTo(x , y + size);
 			
+		ctx.beginPath();
+		ctx.moveTo(x, y);
+		ctx.lineTo(x , y + size);
+				
+		y = y + size;
+		ctx.lineTo(x - size/2 , y + size/2);
 			
+		x = x - size/2;
+		y = y + size/2;
+		ctx.lineTo(x, y - size);
+		ctx.closePath();
+		ctx.fillStyle = color;
+		ctx.stroke();
+		ctx.fill();	
 			
-			y = y + size;
-			ctx.lineTo(x - size/2 , y + size/2);
-			
-			x = x - size/2;
-			y = y + size/2;
-			ctx.lineTo(x, y - size);
-			ctx.closePath();
-			ctx.fillStyle = color;
-			ctx.stroke();
-			ctx.fill();
-			*/
-			
-			
-			
-		}
+		y = y + size;
+		ctx.lineTo(x - size/2 , y + size/2);
 		
-		//reset values
-		document.getElementById("resetShape").reset();
-	
+		x = x - size/2;
+		y = y + size/2;
+		ctx.lineTo(x, y - size);
+		ctx.closePath();
+		ctx.fillStyle = color;
+		ctx.stroke();
+		ctx.fill();
+		*/		
+
+	//for Shape
+	var centerObject = new CenterForShapesAndLight(document.getElementById('shape_x').value, document.getElementById('shape_y').value, document.getElementById('shape_z').value);
+	var radius		 = document.getElementById('size').value;
+	var reflection   = document.getElementById('reflection').value;
+
+	//converting Hex to RGB. 
+	var hexToRGB 	 = hexToRgb(color); 
+	var colorObject  = new colorObj(hexToRGB[0], hexToRGB[1], hexToRGB[2]);
+
+	//dummy value at the moment for imagePlane Class
+	globalWidth 	 = radius;
+	globalHeight	 = radius;
+
+	//object creation for sphere
+	var sphere 		 = new Shape(centerObject, radius, colorObject, reflection, "Sphere");
+	var sphereObject = new SphereObj(sphere);
+
+	//object creation for cube
+	var cube 		 = new Shape(centerObject, radius, colorObject, reflection, "Cube");
+	var cubeObject   = new CubeObj(cube);
+
+	//for Ambient Light
+	var ambientLight 	= new AmbientLight(document.getElementById('brightness').value, document.getElementById('ambient').value);
+	globalAmbientLight  = ambientLight;
+
+   	 if(shape == "Circle") {
+		arrayListForObject.push(sphereObject);
+		
+	} else {
+		arrayListForObject.push(cubeObject);
+	}
+	//reset values
+	document.getElementById("resetShape").reset();
 		
 };
-	
-function clearGrid() {
-	var c = document.getElementById("myCanvas");
-	var ctx = c.getContext("2d");
-	ctx.clearRect(0, 0, c.width, c.height);
-};
+
+function renderShapes() {
+	//generate the JSON file for the form data and send it as HTTP request
+
+	var xhr = new XMLHttpRequest();
+	var url = "url";
+	xhr.open("POST", url, true);
+	xhr.setRequestHeader("Content-type", "application/json");
+	var jsonData = JSON.stringify({"ImagePlane": {"width": globalWidth, "height": globalHeight}, "Scene": {"Object3D": arrayListForObject, "Light": arrayListForLight, "AmbientLight": globalAmbientLight}});

```

Comment:
Hey Srinath, wouldn't it be better, to use an object instead of writing parts of the JSON as a string? Maybe you can create an object called "RayTracingTask"

Reply:
Pushed a new commit. Please check.